youtube_dl/extractor/frontendmaster.py

```diff
@@ -0,0 +1,310 @@
+            if captions.startswith('WEBVTT'):
+                # Assumes captions are in WEBVTT format
+                captions = captions.replace('WEBVTT', '')
+                captions = captions.replace('.', ',')
+        return captions
+
+    def _get_subtitles(self, video_hash, video_id):
+        captions = self._download_webpage(
+            '%s/%s.vtt' % (self._CAPTIONS_BASE, video_hash), video_id,
+            fatal=False)
+        srt_captions = FrontEndMasterIE._convert_subtitles(captions)
+
+        if srt_captions:
+            return {
+                'en': [{
+                    'ext': 'srt',
+                    'data': srt_captions
+                }]
+            }
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
+        course_id = mobj.group('courseid')
+
+        course_json_content = self._download_course(course_id=course_id,
+                                                    url=url)
+
+        # Necessary to get mandatory informations like title and video_url
+        lesson_index = course_json_content.get('lessonSlugs').index(video_id)
+        lesson_hash = course_json_content.get('lessonHashes')[lesson_index]
+        lesson_data = course_json_content.get('lessonData')[lesson_hash]
+        # This is necessary to get the link for the video
+        lesson_source_base = lesson_data['sourceBase']
+
+        lesson_title = lesson_data['title']
+
+        # Some optional fields
+        lesson_description = lesson_data.get('description')
+        lesson_index = lesson_data.get('index')
+        lesson_slug = lesson_data.get('slug')
+        lesson_thumbnail_url = lesson_data.get('thumbnail')
+        lesson_section_elements = course_json_content.get('lessonElements')
+
+        try:
+            course_sections_pairing = self._pair_section_video_element(
+                lesson_section_elements)
+
+            lesson_section = \
+                course_sections_pairing.get(lesson_index)[0]
+
+            lesson_section_number = \
+                course_sections_pairing.get(lesson_index)[1]
+        except Exception:
+            lesson_section = None
+            lesson_section_number = None
+
+        QUALITIES_PREFERENCE = ('low', 'medium', 'high')
+        quality_key = qualities(QUALITIES_PREFERENCE)
+        QUALITIES = {
+            'low': {'width': 480, 'height': 360},
+            'medium': {'width': 1280, 'height': 720},
+            'high': {'width': 1920, 'height': 1080}
+        }
+
+        AllowedQuality = collections.namedtuple('AllowedQuality',
+                                                ['ext', 'qualities'])
+        ALLOWED_QUALITIES = [
+            AllowedQuality('webm', ['low', 'medium', 'high']),
+            AllowedQuality('mp4', ['low', 'medium', 'high'])
+        ]
+
+        cookies = self._get_cookies(self._COOKIES_BASE)
+        cookies_str = ';'.join(['%s=%s' % (cookie.key, cookie.value)
```

Comment:
What are you doing? These cookies are already in session, no need to bother with Cookie header at all.